tests/linalg_test.py

```diff
@@ -1620,6 +1620,55 @@ def testToeplitzConstructionWithKnownCases(self):
       [3, 2, 1, 4]], dtype=np.float32))
 
 
+  @jtu.sample_product(
+    shape=[[2, 1], [1, 2], [2, 5], [5, 2], [4, 20], [20, 4]],
+    dtype=float_types + complex_types)
+  def testNullSpace(self, shape, dtype):
+    rng = jtu.rand_default(self.rng())
+    X = rng(shape, dtype)
+    args_maker = lambda: [X]
+    self._CheckAgainstNumpy(jtu.promote_like_jnp(osp.linalg.null_space),
+                            jsp.linalg.null_space, args_maker)
+
+  @jtu.sample_product(
+    shape=[[2, 1], [1, 2], [2, 5], [5, 2], [4, 20], [20, 4]],
+    dtype=float_types + complex_types,
+    size=[None, 2, 9])
+  def testNullSpaceCompiled(self, shape, dtype, size):
+    # test compiled version
+    rng = jtu.rand_default(self.rng())
+    X = rng(shape, dtype)
+    if size != None:
+      def scipy_fun(x):
+        result = osp.linalg.null_space(x)
+        if size <= result.shape[1]:
+          return result[:, -size:]
+        else:
+          all_zeros = np.zeros((result.shape[0], size), dtype=x.dtype)
+          all_zeros[:, size-result.shape[1]:] = result
+          return all_zeros
+      args_maker = lambda: [X]
+      jnp_fun = partial(jsp.linalg.null_space, size=size)
+      self._CheckAgainstNumpy(scipy_fun, jnp_fun, args_maker, check_dtypes=False)
+      self._CompileAndCheck(jnp_fun, args_maker)
+    else:
+      err, msg = jax.errors.ConcretizationTypeError, r"Abstract tracer value encountered where concrete value is expected"
+      with self.assertRaisesRegex(err, msg):
+        jax.jit(jsp.linalg.null_space, static_argnames=('size'))(X, size=size)
```

Comment:
Nit - this is basically `jit(jnp_fun)`, so we could simplify things by using that here.

Reply:
yup, corrected in the new combined test